server/src/main/java/org/elasticsearch/transport/InboundMessage.java

```diff
@@ -62,6 +62,7 @@ InboundMessage deserialize(BytesReference reference) throws IOException {
                 long requestId = streamInput.readLong();
                 byte status = streamInput.readByte();
                 Version remoteVersion = Version.fromId(streamInput.readInt());
+                streamInput.setVersion(remoteVersion);
```

Comment:
I'm not sure anything really needs to change, but I think we're now setting the version of the input stream multiple times - `decompressingStream()` and `namedWriteableStream()` in this file both set the version after constructing a new wrapper object, but both wrappers delegate version handling to the original StreamInput object anyway. I'm not sure why it wasn't set here in the first place.

Reply:
Thanks for pointing this out. I went ahead and did some cleanup work here.